lib/rest/Twilio.js

```diff
@@ -271,6 +268,44 @@ Twilio.prototype.request = function request(opts) {
   });
 };
 
+/* jshint ignore:start */
+/**
+ * Adds a region and/or edge to a given hostname
+ *
+ * @function getHostname
+ * @memberof Twilio#
+ *
+ * @param {string} hostname - A URI hostname (e.g. api.twilio.com)
+ * @param {string} targetEdge - The targeted edge location (e.g. sydney)
+ * @param {string} targetRegion - The targeted region location (e.g. au1)
+ */
+/* jshint ignore:end */
+Twilio.prototype.getHostname = function getHostname(hostname, targetEdge,
+                                                     targetRegion) {
+  const domain = 'twilio.com';
+  const defaultRegion = 'us1';
+  const prefix = hostname.split('.' + domain)[0];
+  let [product, edge, region] = prefix.split('.');
+
+  if (region) {
+    targetRegion = targetRegion || region;
+  }
+
+  targetRegion = targetRegion || (targetEdge && defaultRegion);
```

Comment:
In the case of `gethostname('api.region.twilio.com', 'edge')` it will return `api.edge.region.twilio.com` instead of `api.edge.us1.twilio.com`. Is this desired?